### PR TITLE
Liquidation list

### LINKED ISSUE #23: liquidation list
Record the liquidation list currently. 

The computation logic can refer to [parallel-finance/liquidation-client](https://github.com/parallel-finance/liquidation-client)

### PR DESCRIPTION
**Pull request by rjman-ljm**
Close #23

### CODE REVIEW

**Comment by rjman-ljm** [comment]
There's still something to do, testing the function.
Because I don't have the right test environment right now.

**Comment by rjman-ljm** [comment]
Considering merge this pr
liquidation list

File: packages/scan/app/scanner/solvers/liquidation.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  borrower: AccountId;
+  liquidity: Liquidity;
+  shortfall: Shortfall;
+  status: boolean;
+}
+
+class LiquidationSolver {
+    private shorfallRecords
+
+    constructor() {
```
Comment (alannotnerd) [inline packages/scan/app/scanner/solvers/liquidation.ts:17]: Default construct could be removed

File: packages/scan/app/scanner/solvers/liquidation.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          // Retrieve only non-zero shortfall
+          status: !accountLiquidity[1].toBn().isZero()
+        }
+
+        return shorfallRecord
+      });
+    }
+}
+
+export const liquidationSolver = new LiquidationSolver()
```
Comment (alannotnerd) [inline packages/scan/app/scanner/solvers/liquidation.ts:63]: also there is an global apipromise singleton exported.

File: packages/scan/app/service.ts [hunk 3]
```diff
@@ -14,13 +14,17 @@ interface ServiceOption {
...
+    logger.debug(`Build Service url:${url}, endpoint:${endpoint}, blockNumber:${blockNumber}`)
     await Api.init(endpoint)
     await Store.init(url)
     this.initBlockHeight = blockNumber
     return new Service()
   }
 
   async run() {
+    // Debug a given block
+    await this.pollBlock(100720)
```
Comment (alannotnerd) [inline packages/scan/app/service.ts:26]: could be removed cause we has an `initBlockHeight` option.

File: packages/scan/app/service.ts [hunk 4]
```diff
@@ -32,11 +36,20 @@ export class Service {
...
         continue
       }
+      
       await scanner.processBlock(hash, blockNumber)
       await store.setLastBlock(blockNumber, hash.toHex())
       logger.debug(`Block#${blockNumber} indexed`)
     }
   }
+  
+  private async pollBlock(height) {
```
Comment (alannotnerd) [inline packages/scan/app/service.ts:46]: Same reason as above comment

File: packages/scan/app/store/index.ts [hunk 5]
```diff
@@ -11,13 +11,13 @@ export class Store {
...
   }
 
-  static async init(url: string) {
+  static async init(url: string): Promise<void> {
     const client = new MongoClient(url)
     await client.connect()
     store = new Store(client)
   }
 
-  getCols<T extends CollectionKey>(key: T) {
```
Comment (alannotnerd) [inline packages/scan/app/store/index.ts:20]: don't change to any

File: packages/scan/app/service.ts [hunk 6]
```diff
@@ -13,16 +13,17 @@ interface ServiceOption {
...
+  static async build({ endpoint, url, blockNumber }: ServiceOption): Promise<Service> {
+    logger.debug(`Build Service url:${url}, endpoint:${endpoint}, blockNumber:${blockNumber}`)
     await Api.init(endpoint)
     await Store.init(url)
     this.initBlockHeight = blockNumber
     return new Service()
   }
 
-  async run() {
+  async run(): Promise<void> {
```
Comment (alannotnerd) [inline packages/scan/app/service.ts:24]: can be infered

File: packages/scan/package.json [hunk 7]
```diff
@@ -1,29 +1,35 @@
...
+        "nodemon": "^2.0.12",
+        "ts-node": "^10.2.1",
+        "typescript": "^4.4.2"
+    },
+    "dependencies": {
+        "@parallel-finance/api": "^1.1.2",
+        "@polkadot/api": "^5.6.2-5",
+        "@polkadot/rpc-core": "^5.6.2-5",
+        "@polkadot/types": "^5.6.2-5",
+        "@types/lodash": "^4.14.171",
```
Comment (alannotnerd) [inline packages/scan/package.json:29]: It's not necessary for production environment.

File: packages/scan/app/service.ts (hunk 6 above)
Comment (rjman-ljm) [reply packages/scan/app/service.ts:24]: eslint required

**Comment by alannotnerd** [comment]
Remove log files.

File: packages/scan/app/model/staker.ts [hunk 8]
```diff
@@ -1,5 +1,8 @@
 import { CollectionCommon } from './common'
 
 export interface Staker extends CollectionCommon {
-  stakers: { [staker: string]: number }
+  stakers: {
+    [staker: string]: number
+  }
+
```
Comment (alannotnerd) [inline packages/scan/app/model/staker.ts:7]: Remove blank lines

File: packages/scan/app/scanner/solvers/liquidation.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
+import { api } from '../../api';
+import '@parallel-finance/types';
+import { AccountId, Liquidity, Shortfall } from '@parallel-finance/types/interfaces';
+import * as lds from 'lodash';
```
Comment (alannotnerd) [inline packages/scan/app/scanner/solvers/liquidation.ts:4]: The community commonly aliases it as '_', maybe we should follow.

File: packages/scan/app/store/index.ts [hunk 10]
```diff
@@ -26,6 +26,10 @@ export class Store {
     return await col.findOne({}, { sort: { blockHeight: -1 } })
   }
 
+  async setLastShortfallRecords(height: number, shortfallRecords: ShortfallRecord[]): Promise<void> {
```
Comment (alannotnerd) [inline packages/scan/app/store/index.ts:29]: Don't need explicit type annotation if there are no ambigous

**Review by alannotnerd — APPROVED** [review]

**Pull request merged**